### DISCUSSION TITLE
解决wing主题笔记图片原图上传，图片过大导致首页加载慢的问题

### DISCUSSION THREAD

**Discussion by moxiaonai** [Tokinx/Wing#104] [投票]
1. 我是从[danilelxp](https://www.danilelxp.com/)这老哥博客一路找到这个主题的，至今自己优化魔改了一些功能
2. 这老哥已经改的非常优秀了，但是他的主页笔记的图片应该都是从手机上传的原图，基本都是几M以上，图多了以后加载很慢
3. 我使用中也遇到了这个问题，所以有了这篇文章

[Wing主题上传前压缩解决图片过大的问题](https://blog.fenewbee.com/wordpress/compress-img.html)

**Comment by Tokinx** [comment]
比较妥当的做法应该是在输出时输出小图片，而不是上传时压缩图片后上传。
Adams有图片裁剪、压缩方法，后面可以考虑移植到Wing上。

主要是考虑到现在大家都4G、5G了，而且静态文件可以使用第三方的对象存储服务，传输速度应该很快了所以没考虑过压缩图片。

**Reply by moxiaonai** [reply]
是的，我理解还是有很大一部分人需要保存原图的，开始的思路也是从存储的位置进行压缩同时保存原图和缩略图，前端列表默认展示缩略图，点击放大的时候再加载原图，可以选择存储在本地或者上传到第三方存储服务

但是最主要确实如我文中所说，我的小水管带宽确实太小了，我使用笔记的大部分场景可能也会是手机端随手发，前端压的话上传也会更快，而我本身确实没有在博客存储原图的诉求，所以这里直接在前端压缩了，也加了一下限制条件，小图的话不会被压缩，大图的话压缩的质量基本也没太大问题

我觉得我目前的这种方案可以作为一种选项，对于像我这种诉求的用户这种可能更合适，对于服务器和经常发图文图片本身价值比较高的的朋友我最前面的方案可能是最优解

**Comment by Tokinx** [comment]
去年已经针对这个问题发布了优化版本，笔记缩略图会优先展示小体积图片
